app/components/selected_users/index.tsx

```diff
@@ -0,0 +1,240 @@
+        },
+        message: {
+            color: changeOpacity(theme.centerChannelColor, 0.6),
+            fontSize: 12,
+            marginRight: 5,
+            marginTop: 10,
+            marginBottom: 2,
+        },
+    };
+});
+
+export default function SelectedUsers({
+    selectedIds,
+    teammateNameDisplay,
+    showToast = false,
+    toastIcon,
+    toastMessage,
+    onPress,
+    onRemove,
+    buttonIcon,
+    buttonText,
+}: Props) {
+    const theme = useTheme();
+    const style = getStyleFromTheme(theme);
+    const isTablet = useIsTablet();
+    const keyboardHeight = useKeyboardHeight();
+    const insets = useSafeAreaInsets();
+
+    const scrollViewHeight = useSharedValue(0);
+    const [isVisible, setIsVisible] = useState(false);
+
+    useEffect(() => {
+        setIsVisible(Boolean(Object.keys(selectedIds).length));
+    }, [selectedIds]);
+
+    const handleOnPress = async () => {
+        onPress();
+    };
+
+    const users = useMemo(() => {
+        const u = [];
+        for (const id of Object.keys(selectedIds)) {
+            if (!selectedIds[id]) {
+                continue;
+            }
+
+            u.push(
+                <SelectedUser
+                    key={id}
+                    user={selectedIds[id]}
+                    teammateNameDisplay={teammateNameDisplay}
+                    onRemove={onRemove}
+                    testID='create_direct_message.selected_user'
+                />,
+            );
+        }
+        return u;
+    }, [selectedIds, teammateNameDisplay, onRemove]);
+
+    const marginBottom = useMemo(() => {
+        if (!keyboardHeight) {
+            return isTablet ? TABLET_MARGIN_BOTTOM : 0;
+        }
+
+        let margin = BOTTOM_TAB_HEIGHT + BUTTON_HEIGHT + insets.bottom;
+        margin += isTablet ? 0 : -10;
+        return margin;
+    }, [keyboardHeight]);
+
+    const totalPanelHeight = useDerivedValue(() => (
+        isVisible ? scrollViewHeight.value + BUTTON_HEIGHT : 0
+    ), [isVisible]);
+
+    const animatedViewStyle = useAnimatedStyle(() => {
+        return {
+            height: withTiming(totalPanelHeight.value, {duration: 200}),
+            marginBottom,
+        };
+    }, [marginBottom]);
+
+    const onLayout = useCallback((e: LayoutChangeEvent) => {
+        scrollViewHeight.value = Math.min(SCROLL_VIEW_MAX_HEIGHT, e.nativeEvent.layout.height);
+    }, []);
+
+    const contents = useMemo(() => (
+        <Animated.View style={[style.container, animatedViewStyle]}>
+            <ScrollView style={{height: scrollViewHeight.value}}>
+                <View
+                    style={style.users}
+                    onLayout={onLayout}
+                >
+                    {users}
+                </View>
+            </ScrollView>
+            <Button
+                onPress={handleOnPress}
+                icon={buttonIcon}
+                text={buttonText}
+            />
+        </Animated.View>
+    ), [users]);
```

Comment:
This should not be needed, since it is just a component, but if you insist in adding it into a `useMemo`, you are missing many dependencies.

That being said, I would just remove the useMemo.